app/controllers/timelogs_controller.rb

```diff
@@ -0,0 +1,61 @@
+class TimelogsController < ApplicationController
+  before_action :load_user
+
+  def index
+    @timelog = Timelog.where(user_id: current_user.id)
+  end
+
+  def create
+    @timelog = @user.timelogs.create(timelog_params)
+    if @timelog.save
+      redirect_to user_timelogs_path(@user)
+    end
+
+  end
+
+  def update
+    @timelog = Timelog.find(params[:id])
+    if params[:timelog][:duration] == ""
+      params[:timelog][:duration] = 0
+    end
+    @timelog.duration = params[:timelog][:duration]
```

Comment:
It is a bad practice to use `params[...]` for setting value. We should create private methods which sanitize values from users.